sdk/core/core/internal/az_precondition_internal.h

```diff
@@ -68,9 +68,15 @@ AZ_NODISCARD AZ_INLINE bool az_span_is_valid(az_span span, int32_t min_size, boo
     return false;
   }
 
-  uint8_t* ptr = az_span_ptr(span);
+  uint8_t* const ptr = az_span_ptr(span);
   int32_t const span_size = az_span_size(span);
 
+  // Can't wrap over the end of the address space
+  if (span_size > (uint8_t*)~0 - ptr)
```

Comment:
Can you add a comment explaining what exactly you are doing here and why it works? For example, why are we negating 0, etc.

Reply:
Maybe it is end of the week, but I struggle to find the words that won't end up repeating the code.
Any more concrete suggestions? :)

Reply:
Commented

Reply:
Completed.